### ISSUE TITLE
When i change stylecop.json to include "xmlHeader": false" and remove XML headers, it gives SA1633 "The file header XML is invalid" warning

### ISSUE THREAD

**Issue by philipag** [DotNetAnalyzers/StyleCopAnalyzers#2470]
SA1633 should not occur with "xmlHeader": false".  Disabling SA1633 causes all header checking to be disabled, but I want the header to be checked as specified by StyleCop docs.

Sample json:

```json
{
  "$schema": "https://raw.githubusercontent.com/DotNetAnalyzers/StyleCopAnalyzers/master/StyleCop.Analyzers/StyleCop.Analyzers/Settings/stylecop.schema.json",
  "settings": {
    "documentationRules": {
      "copyrightText": "Copyright (c) 2013-2017  XYZ LLC."
	  "xmlHeader": false"
	  headerDecoration": "--------------------------------------------------------------------------------------------------------------------"
    }
  }
}
```

Sample CS file:

```csharp
// --------------------------------------------------------------------------------------------------------------------
// Copyright (c) 2013-2017  XYZ LLC.
// --------------------------------------------------------------------------------------------------------------------
```

**Comment by sharwell** [comment]
@philipag Are you sure that your JSON file shown above is correct? If so, there are syntax errors in it which should be resulting in a [SA0002 (Invalid settings file)](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA0002.md) diagnostic.

1. There needs to be a comma at the end of the `copyrightText` and `xmlHeader` lines
2. There should not be a quote after `false`

**Comment by philipag** [comment]
Thanks for catching that. Amazingly in spite of the errors StyleCop was able to parse it correctly (changing the settings in it affected the solution warnings). Anyway, I fixed it but the result is still wrong as per bug report.

**Comment by philipag** [comment]
Actually after fixing the file, the error are changing.  I will report back to see if this fixes things....

**Comment by sharwell** [comment]
@philipag The best way to see if things are fixed is to Rebuild All. The build process always reports the complete set of warnings based on the up-to-date configuration.

**Comment by philipag** [comment]
Now I am getting SA1636.  Is it not acceptable to the have the "// ----" header before the copyright?  When I remove it and only have the one line with the copyright, SA1636 goes away.

The json file is now as follows:

`
{
  "$schema": "https://raw.githubusercontent.com/DotNetAnalyzers/StyleCopAnalyzers/master/StyleCop.Analyzers/StyleCop.Analyzers/Settings/stylecop.schema.json",
  "settings":
  {
    "documentationRules":
	{
      "copyrightText": "Copyright (c) 2013-2017  XYZ LLC.",
	  "xmlHeader": false,
	  "headerDecoration": "--------------------------------------------------------------------------------------------------------------------"
    }
  }
}
`

**Comment by sharwell** [comment]
The `headerDecoration` is only used when XML headers are enabled. If XML headers are disabled, the `copyrightText` property is expected to contain the complete contents of the header, including any decoration lines if you want them.

**Comment by sharwell** [comment]
```
{
  "settings": {
    "documentationRules": {
      "copyrightText": "{decoration}\r\nCopyright (c) 2013-2017  {companyName}.\r\n{decoration}",
      "companyName": "XYZ LLC",
      "variables": {
        "decoration": "--------------------------------------------------------------------------------------------------------------------"
      }
    }
  }
}
```

**Comment by philipag** [comment]
With that it is now back to giving SA1633

**Comment by sharwell** [comment]
@philipag If you apply the code fix, does it still give SA1633?

**Comment by philipag** [comment]
thanks @sharwell that did the trick.  I had to get rid of the carriage returns since my files don't have them.  It would probably be good to make the matching algorithm treat \r\n the same as \n and match to both (since line endings are managed otherwise)

**Comment by sharwell** [comment]
@philipag I forgot we fixed that a long time ago, so everything works if you just put `\n` in the settings file. I filed #2477 to make everything also "just work" if you put `\r\n` in the settings file.
